src/main/resources/db/migration/V2017_12_08_1132__Prepare_response_structure_for_subclassing.sql

```diff
@@ -0,0 +1,36 @@
+-- Add response type property set to full defence for all existing responses
+UPDATE
+  claim
+SET
+  response = JSONB_SET(response, '{responseType}',  '"FULL_DEFENCE"' :: JSONB)
+WHERE response IS NOT NULL;
+
+-- Add full defence type property by mapping existing response type values
+UPDATE
+  claim
+SET
+  response = JSONB_SET(response, '{defenceType}',  '"DISPUTE"' :: JSONB)
+WHERE response IS NOT NULL AND jsonb_extract_path_text(response, 'response') = 'OWE_NONE';
+
+UPDATE
+  claim
+SET
+  response = JSONB_SET(response, '{defenceType}',  '"ALREADY_PAID"' :: JSONB)
+WHERE response IS NOT NULL AND jsonb_extract_path_text(response, 'response') = 'OWE_ALL_PAID_ALL';
+
+-- Remove old response property
+WITH subquery AS (
+    SELECT
+      id,
+      response::JSONB #- '{response}' as migrated_response
```

Comment:
why are you doing this?

Reply:
This is to remove old property with `OWE_NONE` and `OWE_ALL_PAID_ALL` values. These has been migrated to `defenceType`.

Reply:
We had:

```
response: "OWE_NONE"
```

and now we have:

```
responseType: "FULL_DEFENCE"
defenceType: "DISPUTE"
```

Reply:
We had:

```
response: "OWE_ALL_PAID_ALL"
```

and now we have:

```
responseType: "FULL_DEFENCE"
defenceType: "ALREADY_PAID"
```

Reply:
why!!!!
https://github.com/hmcts/cmc-claim-store/blob/master/domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/Response.java#L48
stupid sneaky JsonProperty rename!

Reply:
that confused me lots

Reply:
Agree, that JSON prop rename is rubbish, confused me many time as well.

Reply:
I thought the response was the response but that goes in the defence field...